pkg/rebuild/schema/form.go

```diff
@@ -0,0 +1,112 @@
+package schema
+
+import (
+	"encoding/json"
+	"errors"
+	"net/url"
+	"reflect"
+	"strings"
+)
+
+var (
+	ErrInvalidType      = errors.New("invalid type")
+	ErrUnsupportedField = errors.New("unsupported field")
+	ErrMissingRequired  = errors.New("missing required field")
+)
+
+type fieldOptions struct {
+	name     string
+	required bool
+}
+
+func options(field reflect.StructField) fieldOptions {
+	var opt fieldOptions
+	parts := strings.Split(field.Tag.Get("form"), ",")
+	if opt.name = parts[0]; opt.name == "" {
+		opt.name = strings.ToLower(field.Name)
+	}
+	for _, val := range parts[1:] {
+		if val == "required" {
+			opt.required = true
+		}
+	}
+	return opt
+}
+
+func ToURLValues(in any) (url.Values, error) {
+	tvalue := reflect.ValueOf(in)
+	ttype := tvalue.Type()
+	if ttype.Kind() != reflect.Struct {
+		return nil, ErrInvalidType
+	}
+	v := url.Values{}
+	for i := 0; i < ttype.NumField(); i++ {
+		field, value := ttype.Field(i), tvalue.Field(i)
+		if !field.IsExported() {
+			continue
+		} else if field.Anonymous {
+			return nil, ErrUnsupportedField
+		}
+		opt := options(field)
+		if value.IsZero() {
+			continue
+		}
+		switch field.Type.Kind() {
+		case reflect.String:
+			v.Set(opt.name, value.String())
+		case reflect.Slice:
+			if field.Type.Elem().Kind() == reflect.String {
+				v[opt.name] = value.Interface().([]string)
+				continue
+			}
+			fallthrough
+		default:
+			jsonv, err := json.Marshal(value.Interface())
+			if err != nil {
+				return nil, err
+			}
+			v.Set(opt.name, string(jsonv))
+		}
+	}
+	return v, nil
+}
+
+func FromURLValues(v url.Values, out any) error {
```

Comment:
Swap argument order to match (dest, src)?

Reply:
this is kind of based off of `json.Unmarshal(reader, &out)`. in fact that might be a better name....

Reply:
yeah ok so i actually changed this. Marshal and Unmarshal with a dedicated package gets closer to behavior that should be more familiar.